src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java

```diff
@@ -65,10 +65,21 @@
 	 * @param logger the logger
 	 */
 	public DefaultSerializer(@Nonnull final ILogger logger) {
-		this.logger = Objects.requireNonNull(logger, "parameter logger cannot be null");
-		this.gson = GsonFactory.getGsonInstance(logger);
+		this(false, logger);
 	}
 
+
+    /**
+     * Creates a DefaultSerializer with an option to enable serializing of the null values.
+     *
+     * @param serializeNulls the setting of whether or not to serialize the null values in the JSON object
+     * @param logger         the logger
+     */
+    public DefaultSerializer(@Nonnull final boolean serializeNulls, @Nonnull final ILogger logger) {
```

Comment:
is there any specific reason why the serializeNulls parameters is first and not last? maybe this is a dotnet bias but usually the optional parameters/additional parameters of overloads go at the end.

Reply:
Only reason I went with this ordering was because configuration (serializing nulls) vs logger.  Java doesn't typically care either way which way it goes, but I can flip them if that feels more standard in the MS repos, I don't really have a preference either way. Most Java code typically instantiates its own logger per class anyway, so even having a logger as a constructor parameter was a little bit strange, I'll be honest.

Reply:
I do have a preference for the ordering I'm indicating because when people are writing code, the auto-completion will continue suggesting them the other overloads as they type the first "mandatory" parameters, making discovery easier. But again I might be biased by dotnet/typescript practices.
As I think as the main maintainer for this repo during the last year I've pretty much followed this pattern for my code changes, let's align on that for consistency please.

It's an interesting comment, how would you swap a logger implementation for another one then if your other classes that need the logger are deeply tied to any implementation? how would you pass logger settings? (for the second one probably using some kind of singleton or static factory I'm guessing)

Reply:
>It's an interesting comment, how would you swap a logger implementation for another one then if your other classes that need the logger are deeply tied to any implementation? how would you pass logger settings? (for the second one probably using some kind of singleton or static factory I'm guessing)

Typically, Java uses a single interface at the core level, and then dependencies are pulled in for different implementations. The common one is [Apache's Log4j2](https://logging.apache.org/log4j/2.x/) and then there's a varied amount of implementations that can work with that API.  Core APIs (typically any library) will only include Log4j2 as a compile dependency and then either have optional dependencies to implementations so that they don't get pulled in by default, and then configuration lives an an XML file in the resource path.

Either way, I'll make the changes quick and update the PR with the suggestions that you've made.